File: unit/calculadora_java/src/main/calc/Calculadora.java
package calc;

import java.util.ArrayList;

public class Calculadora {
	private ArrayList<Integer> historico = new ArrayList<>();
	
	public int somar(int a, int b) {
		int resultado = a + b; 
		salvarResultado(resultado);
		return resultado;
	}
	
	public int subtrair(int a, int b) {
		int resultado = a - b;
		salvarResultado(resultado);
		return resultado;
	}
	
	public int multiplicar(int a, int b) {
		int resultado = a * b;
		salvarResultado(resultado);
		return resultado;
	}
	
	public int dividir(int a, int b) {
		int resultado = a / b;
		salvarResultado(resultado);
		return resultado;
	}
	
	private void salvarResultado(int a) {
		this.historico.add(a);
	}
	
	public ArrayList<Integer> lerHistorico() {
		return this.historico;
	}
}

File: unit/calculadora_python/calculadora.py
class Calculadora:
    def __init__(self):
        self.historico = []

    def somar(self, a, b):
        resultado = a + b
        self.gravar(resultado)
        return resultado

    def subtrair(self, a, b):
        resultado = a - b
        self.gravar(resultado)
        return resultado

    def multiplicar(self, a, b):
        resultado = a * b
        self.gravar(resultado)
        return resultado

    def dividir(self, a, b):
        resultado = int(a / b)
        self.gravar(resultado)
        return resultado

    def gravar(self, valor):
        self.historico.append(valor)

    def ler_historico(self):
        return self.historico
